ui/settings.lua

```diff
@@ -147,10 +175,17 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
                 mail.selected_idxs[setting][playername] = fields[setting] == "true"
                 break
             elseif data.type == "string" then
+                mail.selected_idxs[setting][playername] = fields[setting]
                 if data.dataset and fields["dataset_" .. setting] then
                     mail.selected_idxs[setting][playername] = data.dataset[tonumber(fields["dataset_" .. setting])]
+                    mail.show_settings(playername)
+                end
+            elseif data.type == "number" then
+                mail.selected_idxs[setting][playername] = tonumber(fields[setting])
+                if data.dataset and fields["dataset_" .. setting] then
+                    mail.selected_idxs[setting][playername] = data.dataset[tonumber(fields["dataset_" .. setting])]
+                    mail.show_settings(playername)
                 end
```

Comment:
Nitpicking but it seems like interesting things happen when you set a value to `inf`.

Reply:
What should I do to avoid that ?

Reply:
Actually putting `inf` reset to `0`, to that's OK :+1: :heavy_check_mark: